### PR TITLE
Add sentieon dnamodelapply

### PR DESCRIPTION
**Pull request by ramprasadn**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/sentieon/dnamodelapply/main.nf [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+    secret 'SENTIEON_LICENSE_BASE64'
+
+    container 'nf-core/sentieon:202112.06'
+
+    input:
+    tuple val(meta), path(vcf), path(idx)
+    tuple val(meta2), path(fasta)
+    tuple val(meta3), path(fai)
+    path ml_model
```
Comment (asp8200) [inline modules/nf-core/sentieon/dnamodelapply/main.nf:14]: Should there also be a `meta`-field for the `ml_model`?  (I guess so, but I don't know for sure.)

Comment (ramprasadn) [reply modules/nf-core/sentieon/dnamodelapply/main.nf:14]: No harm in adding one I suppose 😄

Comment (asp8200) [reply modules/nf-core/sentieon/dnamodelapply/main.nf]: It seems that you committed the change on the wrong branch 🤔

https://github.com/nf-core/modules/pull/3733

Not that it matters much. I guess can just approve both PRs.

Comment (ramprasadn) [reply modules/nf-core/sentieon/dnamodelapply/main.nf]: That's for dnascope, which also uses ml_model file :)

Comment (asp8200) [reply modules/nf-core/sentieon/dnamodelapply/main.nf]: Ah 😆  my bad

Comment (asp8200) [reply modules/nf-core/sentieon/dnamodelapply/main.nf]: Seems like I can only handle one Sentieon PR at a time 😆

Comment (ramprasadn) [reply modules/nf-core/sentieon/dnamodelapply/main.nf]: No worries 😄

Comment (ramprasadn) [reply modules/nf-core/sentieon/dnamodelapply/main.nf]: On the other hand, I have been bombarding you with PR reviews 😆

Comment (asp8200) [reply modules/nf-core/sentieon/dnamodelapply/main.nf]: You are on a good roll 😁 👏

File: tests/modules/nf-core/sentieon/dnamodelapply/main.nf [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+    input = [
+        [ id:'test', single_end:false ], // meta map
+        file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true),
+        file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_bam_bai'], checkIfExists: true)
+    ]
+
+    ml_model = [[:], file("https://s3.amazonaws.com/sentieon-release/other/SentieonDNAscopeModel1.0.model", checkIfExists: true)]
+
+    SENTIEON_DNASCOPE ( input, fasta, fai, [[:],[]], [[:],[]], [[:],[]], ml_model )
```
Comment (asp8200) [inline tests/modules/nf-core/sentieon/dnamodelapply/main.nf:21]:
```suggestion
    SENTIEON_DNASCOPE ( input, fasta, fai, [[:],[]], [[:],[]], [[:],[]], [[:],ml_model] )
```

File: tests/modules/nf-core/sentieon/dnamodelapply/nextflow.config [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+    publishDir = { "${params.outdir}/${task.process.tokenize(':')[-1].tokenize('_')[0].toLowerCase()}" }
+
+    withLabel: 'sentieon' {
+        ext.sentieon_auth_mech_base64 = secrets.SENTIEON_AUTH_MECH_BASE64
+        ext.sentieon_auth_data_base64 = secrets.SENTIEON_AUTH_DATA_BASE64
+    }
+
+    withName: 'SENTIEON_DNASCOPE' {
+        ext.prefix = 'dnascope'
```
Comment (asp8200) [inline tests/modules/nf-core/sentieon/dnamodelapply/nextflow.config:11]: why is that needed?

File: tests/modules/nf-core/sentieon/dnamodelapply/main.nf [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    fasta = Channel.of([ [ id:'test' ], file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true)]).collect()
+    fai   = Channel.of([ [ id:'test' ], file(params.test_data['homo_sapiens']['genome']['genome_fasta_fai'], checkIfExists: true)]).collect()
+
+    input = [
+        [ id:'test', single_end:false ], // meta map
+        file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true),
+        file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_bam_bai'], checkIfExists: true)
+    ]
+
+    ml_model = [[:], file("https://s3.amazonaws.com/sentieon-release/other/SentieonDNAscopeModel1.0.model", checkIfExists: true)]
```
Comment (asp8200) [inline tests/modules/nf-core/sentieon/dnamodelapply/main.nf:19]: Isn't there a model 1.1 around?

Comment (asp8200) [reply tests/modules/nf-core/sentieon/dnamodelapply/main.nf:19]: I got `SentieonDNAscopeModel1.1.model` :-)

Comment (asp8200) [reply tests/modules/nf-core/sentieon/dnamodelapply/main.nf:19]:
```suggestion
    ml_model = [[:], file("https://s3.amazonaws.com/sentieon-release/other/SentieonDNAscopeModel1.1.model", checkIfExists: true)]
```

**Review by asp8200 — APPROVED** [review]
Looks good 👏
